src/Core/Table/CompositeHeader.fs

```diff
@@ -432,6 +432,15 @@ type CompositeHeader =
         | ProtocolType | ProtocolREF | ProtocolDescription | ProtocolUri | ProtocolVersion | Performer | Date | Input _ | Output _ -> true
         | _ -> false
 
+    member this.Copy() =
+        match this with
+        | Parameter oa -> Parameter (oa.Copy())
+        | Factor oa -> Factor (oa.Copy())
+        | Characteristic oa -> Characteristic (oa.Copy())
+        | Component oa -> Component (oa.Copy())
```

Comment:
What happens to Input/Output of Data?

Reply:
Ahh the actual data entity is stored in the cells, not the header. So this is fine anyways!

Reply:
Added some tests for data cell. Also, There already was one test which checked updating an "Input [Sample]" column to an "Input [Data]" Column.